### PR TITLE
DI framework for ios

### PR DESCRIPTION
**Pull request by watanavex**
## Issue
- #353

## Overview (Required)
- I adopted Koin
- Android targets keeps Dagger Hilt
- This PR, DB and API modules are supported
- Repository module has another issue (#395), I'll support with it later

## Links
- https://insert-koin.io

## Screenshot
🙅‍♂️

### CODE REVIEW

File: data/api/src/iosMain/kotlin/io/github/droidkaigi/feeder/data/request/DevicePostRequest.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package io.github.droidkaigi.feeder.data.request
+
+actual fun platform() = "ios"
```
Comment (watanavex) [inline data/api/src/iosMain/kotlin/io/github/droidkaigi/feeder/data/request/DevicePostRequest.kt:3]: Is this correct?

**Comment by takahirom** [comment]
👀

**Comment by takahirom** [comment]
It's nice that koin's API is easy to understand!

File: data/api/build.gradle [hunk 2]
```diff
@@ -19,7 +19,13 @@ tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile.class).all {
...
 kotlin {
     android()
-    ios()
+
+    // Add after resolving Firebase auth dependencies
+    // if (System.getenv("SDK_NAME")?.startsWith("iphoneos")) {
+    //     iosArm64("ios")
+    // } else {
+    //     iosX64("ios")
+    // }
```
Comment (watanavex) [inline data/api/build.gradle:28]: I'm having a hard time linking Firebase auth 🤔 
Therefore, it has not been tested with the API module🙇‍♂️
In this PR, I will stop adding the iOS target to the API module, and would like to support it in another PR🙇‍♂️

Comment (takahirom) [reply data/api/build.gradle:28]: I also had a hard time linking this library before. 👍

**Review by takahirom — APPROVED** [review]

**Comment by takahirom** [comment]
LGTM 👍

**Pull request merged**